src/tqec/block/block.py

```diff
@@ -295,20 +316,22 @@ def to_circuit(self) -> cirq.Circuit:
             a cirq.Circuit instance representing the full computation.
         """
         instantiated_scheduled_blocks: list[ScheduledCircuit] = []
+        depth = 0
         for position, block in self.blocks.items():
-
-            def shift_qubits(q: cirq.Qid) -> cirq.GridQubit:
-                if not isinstance(q, cirq.GridQubit):
-                    raise TQECException(
-                        f"Found a circuit with {q} that is not a cirq.GridQubit instance."
-                    )
-                return q + (position.x, position.y)
-
             spatially_shifted_circuit = block.instantiate().transform_qubits(
-                shift_qubits
+                partial(_shift_qubits, position)
             )
             instantiated_scheduled_blocks.append(
-                ScheduledCircuit(spatially_shifted_circuit, position.z)
+                ScheduledCircuit(spatially_shifted_circuit, depth)
             )
+            depth += block.depth
```

Comment:
This will need to be improved as it will break for any computation containing more than 1 qubit. I think the best way of getting the depth each block should be located would be to perform an exploration of the graph-like representation of the computation, starting from the `t=0` side, compute the depth by exploring the graph, and raise in the process if any incompatibility is found. Something very similar to the `ComposedTemplate` internals, using a DFS exploration, would work in my opinion.

Reply:
This is similar to what I had in mind; I just started with the happy case.
Could two qubits have cubes of different sizes in a case we should be able to treat? I assumed that if we had the same code, each repetition would have the same number of moments. Unless the user specifies a case where we have a different number of repetitions, the blocks should be all equal. If they are not equal I would then raise a warning to the user.

Reply:
Is there use cases where two cubes can have any inconsistent dimension? In my mind, all the computation should admit an atomic cube definition, at least for our current workflow.